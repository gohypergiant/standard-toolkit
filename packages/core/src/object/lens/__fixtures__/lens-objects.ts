export type City = 'New York' | 'San Diego' | 'Austin';

export type Address = {
  city: City;
  street: string;
};

export type Person = {
  name: string;
  address?: Address;
};

export type Profile = {
  info: Person;
};

export type User = {
  username: string;
  password: string;
  profile: Profile[];
};

export const personStore: Person = {
  name: 'Alice',
  address: {
    city: 'New York',
    street: '123 Main St',
  },
};

export const userStore: User = {
  username: 'foobar',
  password: 'cleverPassword',
  profile: [
    {
      info: {
        name: 'Frank',
        address: {
          city: 'Austin',
          street: '123 Main St',
        },
      },
    },
  ],
};
