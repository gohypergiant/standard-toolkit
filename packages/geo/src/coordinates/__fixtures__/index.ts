/*
 * Copyright 2024 Hypergiant Galactic Systems Inc. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */

// const LAT_DD = 40.689247;
// const LON_DD = -74.044502;

export const ddPairs = [
  ['partial', '40, -74'],
  ['partial directions prefix', 'N 40, W 74'],
  ['partial directions prefix no space', 'N40, W74'],
  ['partial directions suffix', '40 N, 74 W'],
  ['partial directions suffix no space', '40N, 74W'],
  ['plus minus', '+40.689247, -74.044502'],
  ['inferred plus', '40.689247, -74.044502'],
  ['inferred positive direction', '40.689247, 74.044502 W'],
  ['degree delimeters', '40.689247° N, 74.044502° W'],
  ['separator ,', '40.689247,-74.044502'],
  ['separator /', '40.689247/-74.044502'],
  ['separator space', '40.689247 -74.044502'],
  ['NSEW directions prefix', 'N 40.689247, W 74.044502'],
  ['NSEW directions prefix no space', 'N40.689247, W74.044502'],
  ['NSEW directions suffix', '40.689247 N, 74.044502 W'],
  ['NSEW directions suffix no space', '40.689247N, 74.044502W'],
  ['mixed plus minus and directions', '+40.689247, W74.044502'],
  ['mixed full and partial', '40.689247, W74'],
  ['odd whitespace', ' + 40.689247  ,   -  74.044502    '],
];

// const LAT_DMS = `40° 41' 21.2892"`;
// const LON_DMS = `-74° 02' 40.2066"`;

export const dmsPairs = [
  ['partial no delimeters', '40 41 -74 02'],
  ['partial delimeters', `40° 41' -74 02'`],
  ['plus minus no delimeters', '+40 41 21.2892 -74 02 40.2066'],
  ['plus minus delimeters', `+40° 41' 21.2892" -74° 02' 40.2066"`],
  ['directions no delimeters', '40 41 21.2892 N 74 02 40.2066 W'],
  ['directions delimeters', `40° 41' 21.2892" N 74° 02' 40.2066" W`],
  ['separator ,', `40° 41' 21.2892",-74° 02' 40.2066"`],
  ['separator /', `40° 41' 21.2892"/-74° 02' 40.2066"`],
  ['separator space', `40° 41' 21.2892" -74° 02' 40.2066"`],
];

export const ddmPairs = [];

export const mixedPairs = [];
