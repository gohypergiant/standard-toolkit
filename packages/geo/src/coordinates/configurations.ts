// __private-exports

import { DD_LAT, DD_LON, SEPARATORS, DMS_LAT, DMS_LON } from './regex';

/**
 * NOTE: these pairs are setup to support different lon/lat ordering
 * 0 = latitude first, 1 = longitude first
 */

export const dd = [
  new RegExp(DD_LAT + SEPARATORS + DD_LON, 'i'),
  new RegExp(DD_LON + SEPARATORS + DD_LAT, 'i'),
] as const;

export const dms = [
  new RegExp(DMS_LAT + SEPARATORS + DMS_LON, 'i'),
  new RegExp(DMS_LON + SEPARATORS + DMS_LAT, 'i'),
] as const;
