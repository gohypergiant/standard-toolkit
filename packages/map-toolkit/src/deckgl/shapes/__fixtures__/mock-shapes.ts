/*
 * Copyright 2025 Hypergiant Galactic Systems Inc. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */

import { uuid } from '@accelint/core';
import type { EditableShape } from '../shared/types';

/**
 * Mock shapes for Storybook examples
 * Centered around the East Coast of the United States for better visibility
 */
export const mockShapes: EditableShape[] = [
  // Circle shape
  {
    id: uuid(),
    name: 'New Circle',
    label: 'Circle',
    locked: false,
    shapeType: 'Circle',
    feature: {
      type: 'Feature',
      properties: {
        editProperties: {
          radius: {
            value: 250,
            units: 'kilometers',
          },
          center: [-82.16095, 41.459647],
        },
        styleProperties: {
          fillColor: '#62a6ff',
          strokeColor: '#62a6ff',
          strokeWidth: 4,
          fillOpacity: 100,
          strokeOpacity: 100,
          strokePattern: 'solid',
        },
        shapeId: uuid(),
      },
      geometry: {
        type: 'Polygon',
        coordinates: [
          [
            [-82.16095, 43.707947909311336],
            [-82.36015690912059, 43.70316028540156],
            [-82.55845254686115, 43.68881937491615],
            [-82.75493115613983, 43.66499092635894],
            [-82.94869794613852, 43.6317840714271],
            [-83.13887442101594, 43.589350656918235],
            [-83.32460352721753, 43.53788432050035],
            [-83.50505456526025, 43.47761932112011],
            [-83.67942781700711, 43.40882913745322],
            [-83.84695884550976, 43.331824850117954],
            [-84.00692243127878, 43.24695332533256],
            [-84.15863611611054, 43.154595219270426],
            [-84.30146333312614, 43.05516282353598],
            [-84.43481610923294, 42.94909777294539],
            [-84.55815733358935, 42.8368686371589],
            [-84.67100259264551, 42.71896841769618],
            [-84.7729215787836, 42.595911971503746],
            [-84.86353908536248, 42.46823338157134],
            [-84.94253560598052, 42.33648329415702],
            [-85.00964755995483, 42.20122624102555],
            [-85.06466716933859, 42.063037963780396],
            [-85.1074420152706, 41.922502755924675],
            [-85.13787430309867, 41.78021083676712],
            [-85.15591986659554, 41.63675576973743],
            [-85.16158694176136, 41.49273193613174],
            [-85.15493474026309, 41.34873207380365],
            [-85.13607185158772, 41.20534488887921],
            [-85.10515450157664, 41.063152747226816],
            [-85.06238469325403, 40.9227294511701],
            [-85.00800825385319, 40.78463810580728],
            [-84.9423128097613, 40.64942907829733],
            [-84.86562570882357, 40.51763805259652],
            [-84.77831190713269, 40.38978418137459],
            [-84.68077183514008, 40.26636833620473],
            [-84.57343925570261, 40.14787145659815],
            [-84.45677912456257, 40.034752998033355],
            [-84.33128546177456, 39.9274494788032],
            [-84.1974792407598, 39.826373125257724],
            [-84.05590630000027, 39.73191061484778],
            [-83.90713528088291, 39.64442191626177],
            [-83.75175559387282, 39.564239225887086],
            [-83.59037541402563, 39.49166599980656],
            [-83.42361970583754, 39.42697608055334],
            [-83.25212827656625, 39.37041291788369],
            [-83.07655385642542, 39.32218888288349],
            [-82.89756020344666, 39.28248467479111],
            [-82.71582023030466, 39.251448819996405],
            [-82.53201414999904, 39.22919726275635],
            [-82.34682763697197, 39.21581304725106],
            [-82.16095, 39.21134609068866],
            [-81.97507236302806, 39.21581304725106],
            [-81.78988585000097, 39.22919726275635],
            [-81.60607976969536, 39.251448819996405],
            [-81.42433979655335, 39.28248467479111],
            [-81.2453461435746, 39.32218888288349],
            [-81.06977172343376, 39.37041291788369],
            [-80.89828029416246, 39.42697608055334],
            [-80.73152458597438, 39.49166599980656],
            [-80.57014440612718, 39.564239225887086],
            [-80.4147647191171, 39.64442191626177],
            [-80.26599369999974, 39.73191061484778],
            [-80.12442075924021, 39.826373125257724],
            [-79.99061453822544, 39.9274494788032],
            [-79.86512087543743, 40.034752998033355],
            [-79.74846074429742, 40.14787145659815],
            [-79.64112816485994, 40.26636833620473],
            [-79.54358809286732, 40.38978418137459],
            [-79.45627429117646, 40.51763805259652],
            [-79.37958719023872, 40.64942907829733],
            [-79.31389174614682, 40.78463810580728],
            [-79.25951530674598, 40.922729451170106],
            [-79.21674549842339, 41.063152747226816],
            [-79.18582814841231, 41.20534488887921],
            [-79.16696525973693, 41.34873207380365],
            [-79.16031305823866, 41.49273193613174],
            [-79.16598013340447, 41.63675576973741],
            [-79.18402569690133, 41.78021083676712],
            [-79.21445798472942, 41.922502755924675],
            [-79.25723283066142, 42.063037963780396],
            [-79.31225244004519, 42.20122624102555],
            [-79.3793643940195, 42.33648329415702],
            [-79.45836091463754, 42.46823338157134],
            [-79.5489784212164, 42.595911971503746],
            [-79.65089740735452, 42.71896841769618],
            [-79.76374266641066, 42.8368686371589],
            [-79.88708389076709, 42.94909777294539],
            [-80.02043666687388, 43.05516282353598],
            [-80.16326388388947, 43.154595219270426],
            [-80.31497756872125, 43.24695332533256],
            [-80.47494115449025, 43.331824850117954],
            [-80.6424721829929, 43.40882913745322],
            [-80.81684543473976, 43.47761932112011],
            [-80.9972964727825, 43.53788432050035],
            [-81.18302557898407, 43.589350656918235],
            [-81.37320205386149, 43.6317840714271],
            [-81.5669688438602, 43.66499092635894],
            [-81.76344745313887, 43.68881937491615],
            [-81.96174309087944, 43.70316028540156],
            [-82.16095, 43.707947909311336],
          ],
        ],
      },
    },
    lastUpdated: 1761946918381,
  },
  // LineString shape
  {
    id: uuid(),
    name: 'New LineString',
    label: 'LineString',
    locked: false,
    shapeType: 'LineString',
    feature: {
      type: 'Feature',
      properties: {
        styleProperties: {
          fillColor: '#62a6ff',
          strokeColor: '#30D27E',
          strokeWidth: 4,
          fillOpacity: 100,
          strokeOpacity: 100,
          strokePattern: 'dashed',
        },
        shapeId: uuid(),
      },
      geometry: {
        type: 'LineString',
        coordinates: [
          [-68.72187009616401, 42.71503842674323],
          [-74.46583143039219, 39.27006247796291],
          [-69.18729871382442, 36.97779398594955],
          [-65.15565961470328, 39.49032120221301],
          [-61.8594178544504, 35.39820577380118],
        ],
      },
    },
    lastUpdated: 1761946887453,
  },
  // Point shape
  {
    id: uuid(),
    name: 'New Point',
    label: 'Point',
    locked: false,
    shapeType: 'Point',
    feature: {
      type: 'Feature',
      properties: {
        styleProperties: {
          fillColor: '#62a6ff',
          strokeColor: '#62a6ff',
          strokeWidth: 4,
          fillOpacity: 100,
          strokeOpacity: 100,
          strokePattern: 'solid',
        },
        shapeId: uuid(),
      },
      geometry: {
        type: 'Point',
        coordinates: [-79.35476906757644, 34.81657315443721],
      },
    },
    lastUpdated: 1761946839060,
  },
  // Polygon shape
  {
    id: uuid(),
    name: 'New Polygon',
    label: 'Polygon',
    locked: false,
    shapeType: 'Polygon',
    feature: {
      type: 'Feature',
      properties: {
        styleProperties: {
          fillColor: '#D4231D',
          strokeColor: '#62a6ff',
          strokeWidth: 4,
          fillOpacity: 100,
          strokeOpacity: 100,
          strokePattern: 'dotted',
        },
        shapeId: uuid(),
      },
      geometry: {
        type: 'Polygon',
        coordinates: [
          [
            [-77.37043805542108, 33.36809986665163],
            [-77.98079140644158, 30.023275562140917],
            [-72.62863583984473, 30.135034684161678],
            [-71.7100758117689, 33.44562505532675],
            [-73.94373364868278, 35.21926062000649],
            [-77.37043805542108, 33.36809986665163],
          ],
        ],
      },
    },
    lastUpdated: 1761946937125,
  },
];
