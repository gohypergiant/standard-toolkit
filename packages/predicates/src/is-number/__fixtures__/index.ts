/*
 * Copyright 2024 Hypergiant Galactic Systems Inc. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */

export const numberLiteralPairs = [
  ['negative integer number', -1],
  ['zero integer number', 0],
  ['positive integer number', 1],
  ['octal integer literal', 0o0144],
  ['hexadecimal integer literal', 0xfff],
] as const;

export const floatingPointPairs = [
  ['nagative floating point number', -1.1234],
  ['positive floating point number', 1.1234],
  ['exponential notation', 8e5],
] as const;

export const numberLikeStringPairs = [
  ['negative integer string', '-1'],
  ['zero string', '0'],
  ['positive integer string', '1'],
  ['octal integer literal string', '040'],
  ['hexadecimal integer literal string', '0xFF'],
  ['zero padded integer string', '05'],
] as const;

export const floatingPointStringPairs = [
  ['negative floating point string', '-1.1234'],
  ['positive floating point string', '1.1234'],
  ['exponential notation string', '123e-2'],
  ['zero padded positive floating point string', '01.1234'],
] as const;

export const nonFinitePairs = [
  ['NaN', Number.NaN],
  ['positive Infinity', Number.POSITIVE_INFINITY],
  ['negative Infinity', Number.NEGATIVE_INFINITY],
  // biome-ignore lint/style/useNumberNamespace: want to diff against Number.POSITIVE_INFINITY
  ['positive Infinity primitive', Infinity],
  // biome-ignore lint/style/useNumberNamespace: want to diff against Number.NEGATIVE_INFINITY
  ['negative Infinity primative', -Infinity],
];

export const nonFiniteStringPairs = [
  ['NaN string', 'NaN'],
  ['positive Infinity string', 'Infinity'],
  ['negative Infinity string', '-Infinity'],
] as const;

export const nonNumericPairs = [
  ['empty string', ''],
  ['whitespace character string', '   '],
  ['tab character string', '\t\t'],
  ['alphanumeric character string', 'abcdefghijklm1234567890'],
  ['non-numerc character string', 'xabcdefx'],
  ['number with preceding non-numeric characters', 'bcfed5.2'],
  ['number with trailing non-numeric characters', '7.2acdgs'],
  ['boolean true literal', true],
  ['boolean false literal', false],
  ['undefined value', undefined],
  ['null value', null],
  ['date object value', new Date()],
  ['object value', {}],
  [
    'function instance',
    () => {
      // noop
    },
  ],
] as const;
