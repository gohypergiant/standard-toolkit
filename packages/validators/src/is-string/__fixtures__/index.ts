/*
 * Copyright 2024 Hypergiant Galactic Systems Inc. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */

export const nonStringPairs = [
  ['undefined', undefined],
  ['null', null],
  ['false', false],
  ['true', true],
  ['array', []],
  ['object', {}],
  [
    'function',
    () => {
      // noop
    },
  ],
  ['regex literal', /a/g],
  ['date object', new Date()],
  ['number', 1],
  ['number object', Object(1)],
  ['NaN', Number.NaN],
  ['negative infinity', Number.NEGATIVE_INFINITY],
  ['positive infinity', Number.POSITIVE_INFINITY],
] as const;
