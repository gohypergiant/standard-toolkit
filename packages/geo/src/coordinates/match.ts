/*
 * Copyright 2024 Hypergiant Galactic Systems Inc. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */

import { dd, dms } from './configurations';

type Nullish<T> = T | undefined | null | '';
type PlusMinus = '+' | '-';
type Directions = 'N' | 'S' | 'E' | 'W';

// biome-ignore lint/style/useNamingConvention: <explanation>
type DDMatches = [
  Nullish<Directions>,
  Nullish<PlusMinus>,
  Nullish<string>, // degrees
  Nullish<string>, // decimals
  Nullish<Directions>,
  Nullish<Directions>,
  Nullish<PlusMinus>,
  Nullish<string>, // degrees
  Nullish<string>, // decimals
  Nullish<Directions>,
];

const ddFallback = Array.from(Array(10), () => '');

// biome-ignore lint/style/useNamingConvention: <explanation>
export function matchDD(val: unknown): DDMatches {
  // TODO: lat/lon ordering

  const matches = `${val}`.match(dd[0]);

  if (!matches) {
    return ddFallback as DDMatches;
  }

  // Remove first regex.match entry since it is the input value
  return Array.from(matches).slice(1) as DDMatches;
}

// biome-ignore lint/style/useNamingConvention: <explanation>
type DMSMatches = [
  Nullish<Directions>,
  Nullish<PlusMinus>,
  Nullish<string>, // degrees
  Nullish<string>, // minutes
  Nullish<string>, // seconds
  Nullish<string>, // milliarcseconds
  Nullish<Directions>,
  Nullish<Directions>,
  Nullish<PlusMinus>,
  Nullish<string>, // degrees
  Nullish<string>, // minutes
  Nullish<string>, // seconds
  Nullish<string>, // milliarcseconds
  Nullish<Directions>,
];

const dmsFallback = Array.from(Array(14), () => '');

// biome-ignore lint/style/useNamingConvention: <explanation>
export function matchDMS(val: unknown): DMSMatches {
  // TODO: lat/lon ordering

  const matches = `${val}`.match(dms[0]);

  if (!matches) {
    return dmsFallback as DMSMatches;
  }

  // Remove first regex.match entry since it is the input value
  return Array.from(matches).slice(1) as DMSMatches;
}
