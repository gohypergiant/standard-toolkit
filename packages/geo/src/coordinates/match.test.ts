/*
 * Copyright 2024 Hypergiant Galactic Systems Inc. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */

import { expect, it, describe } from 'vitest';
import { dd, dms } from './configurations';
import { ddPairs, dmsPairs } from './__fixtures__';
import { matchDD, matchDMS } from './match';

console.log(dd[0]);
console.log(dms[0]);

describe('coordinates', () => {
  describe('matching', () => {
    describe('decimal degrees', () => {
      for (const pairs of ddPairs) {
        it(`${pairs[0]}: ${pairs[1]}`, () => {
          const matches = matchDD(pairs[1]);

          expect(matches).toMatchSnapshot();
        });
      }
    });
  });

  describe('degrees minutes seconds', () => {
    for (const pairs of dmsPairs) {
      it(`${pairs[0]}: ${pairs[1]}`, () => {
        const matches = matchDMS(pairs[1]);

        expect(matches).toMatchSnapshot();
      });
    }
  });
});
