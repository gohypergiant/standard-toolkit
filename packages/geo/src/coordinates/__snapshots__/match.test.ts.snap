// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`coordinates > degrees minutes seconds > directions delimeters: 40° 41' 21.2892" N 74° 02' 40.2066" W 1`] = `
[
  "",
  "",
  "40",
  "41",
  "21",
  ".2892",
  "N",
  "",
  "",
  "74",
  "02",
  "40",
  ".2066",
  "W",
]
`;

exports[`coordinates > degrees minutes seconds > directions no delimeters: 40 41 21.2892 N 74 02 40.2066 W 1`] = `
[
  "",
  "",
  "40",
  "41",
  "21",
  ".2892",
  "N",
  "",
  "",
  "74",
  "02",
  "40",
  ".2066",
  "W",
]
`;

exports[`coordinates > degrees minutes seconds > partial delimeters: 40° 41' -74 02' 1`] = `
[
  "",
  "",
  "",
  "",
  "",
  "",
  "",
  "",
  "",
  "",
  "",
  "",
  "",
  "",
]
`;

exports[`coordinates > degrees minutes seconds > partial no delimeters: 40 41 -74 02 1`] = `
[
  "",
  "",
  "",
  "",
  "",
  "",
  "",
  "",
  "",
  "",
  "",
  "",
  "",
  "",
]
`;

exports[`coordinates > degrees minutes seconds > plus minus delimeters: +40° 41' 21.2892" -74° 02' 40.2066" 1`] = `
[
  "",
  "+",
  "40",
  "41",
  "21",
  ".2892",
  "",
  "",
  "-",
  "74",
  "02",
  "40",
  ".2066",
  "",
]
`;

exports[`coordinates > degrees minutes seconds > plus minus no delimeters: +40 41 21.2892 -74 02 40.2066 1`] = `
[
  "",
  "+",
  "40",
  "41",
  "21",
  ".2892",
  "",
  "",
  "-",
  "74",
  "02",
  "40",
  ".2066",
  "",
]
`;

exports[`coordinates > degrees minutes seconds > separator ,: 40° 41' 21.2892",-74° 02' 40.2066" 1`] = `
[
  "",
  "",
  "40",
  "41",
  "21",
  ".2892",
  "",
  "",
  "-",
  "74",
  "02",
  "40",
  ".2066",
  "",
]
`;

exports[`coordinates > degrees minutes seconds > separator /: 40° 41' 21.2892"/-74° 02' 40.2066" 1`] = `
[
  "",
  "",
  "40",
  "41",
  "21",
  ".2892",
  "",
  "",
  "-",
  "74",
  "02",
  "40",
  ".2066",
  "",
]
`;

exports[`coordinates > degrees minutes seconds > separator space: 40° 41' 21.2892" -74° 02' 40.2066" 1`] = `
[
  "",
  "",
  "40",
  "41",
  "21",
  ".2892",
  "",
  "",
  "-",
  "74",
  "02",
  "40",
  ".2066",
  "",
]
`;

exports[`coordinates > matching > decimal degrees > NSEW directions prefix no space: N40.689247, W74.044502 1`] = `
[
  "N",
  "",
  "40",
  ".689247",
  "",
  "W",
  "",
  "74",
  ".044502",
  "",
]
`;

exports[`coordinates > matching > decimal degrees > NSEW directions prefix: N 40.689247, W 74.044502 1`] = `
[
  "N",
  "",
  "40",
  ".689247",
  "",
  "W",
  "",
  "74",
  ".044502",
  "",
]
`;

exports[`coordinates > matching > decimal degrees > NSEW directions suffix no space: 40.689247N, 74.044502W 1`] = `
[
  "",
  "",
  "40",
  ".689247",
  "N",
  "",
  "",
  "74",
  ".044502",
  "W",
]
`;

exports[`coordinates > matching > decimal degrees > NSEW directions suffix: 40.689247 N, 74.044502 W 1`] = `
[
  "",
  "",
  "40",
  ".689247",
  "N",
  "",
  "",
  "74",
  ".044502",
  "W",
]
`;

exports[`coordinates > matching > decimal degrees > degree delimeters: 40.689247° N, 74.044502° W 1`] = `
[
  "",
  "",
  "40",
  ".689247",
  "N",
  "",
  "",
  "74",
  ".044502",
  "W",
]
`;

exports[`coordinates > matching > decimal degrees > inferred plus: 40.689247, -74.044502 1`] = `
[
  "",
  "",
  "40",
  ".689247",
  "",
  "",
  "-",
  "74",
  ".044502",
  "",
]
`;

exports[`coordinates > matching > decimal degrees > inferred positive direction: 40.689247, 74.044502 W 1`] = `
[
  "",
  "",
  "40",
  ".689247",
  "",
  "",
  "",
  "74",
  ".044502",
  "W",
]
`;

exports[`coordinates > matching > decimal degrees > mixed full and partial: 40.689247, W74 1`] = `
[
  "",
  "",
  "40",
  ".689247",
  "",
  "W",
  "",
  "74",
  undefined,
  "",
]
`;

exports[`coordinates > matching > decimal degrees > mixed plus minus and directions: +40.689247, W74.044502 1`] = `
[
  "",
  "+",
  "40",
  ".689247",
  "",
  "W",
  "",
  "74",
  ".044502",
  "",
]
`;

exports[`coordinates > matching > decimal degrees > odd whitespace:  + 40.689247  ,   -  74.044502     1`] = `
[
  "",
  "+",
  "40",
  ".689247",
  "",
  "",
  "-",
  "74",
  ".044502",
  "",
]
`;

exports[`coordinates > matching > decimal degrees > partial directions prefix no space: N40, W74 1`] = `
[
  "N",
  "",
  "40",
  undefined,
  "",
  "W",
  "",
  "74",
  undefined,
  "",
]
`;

exports[`coordinates > matching > decimal degrees > partial directions prefix: N 40, W 74 1`] = `
[
  "N",
  "",
  "40",
  undefined,
  "",
  "W",
  "",
  "74",
  undefined,
  "",
]
`;

exports[`coordinates > matching > decimal degrees > partial directions suffix no space: 40N, 74W 1`] = `
[
  "",
  "",
  "40",
  undefined,
  "N",
  "",
  "",
  "74",
  undefined,
  "W",
]
`;

exports[`coordinates > matching > decimal degrees > partial directions suffix: 40 N, 74 W 1`] = `
[
  "",
  "",
  "40",
  undefined,
  "N",
  "",
  "",
  "74",
  undefined,
  "W",
]
`;

exports[`coordinates > matching > decimal degrees > partial: 40, -74 1`] = `
[
  "",
  "",
  "40",
  undefined,
  "",
  "",
  "-",
  "74",
  undefined,
  "",
]
`;

exports[`coordinates > matching > decimal degrees > plus minus: +40.689247, -74.044502 1`] = `
[
  "",
  "+",
  "40",
  ".689247",
  "",
  "",
  "-",
  "74",
  ".044502",
  "",
]
`;

exports[`coordinates > matching > decimal degrees > separator ,: 40.689247,-74.044502 1`] = `
[
  "",
  "",
  "40",
  ".689247",
  "",
  "",
  "-",
  "74",
  ".044502",
  "",
]
`;

exports[`coordinates > matching > decimal degrees > separator /: 40.689247/-74.044502 1`] = `
[
  "",
  "",
  "40",
  ".689247",
  "",
  "",
  "-",
  "74",
  ".044502",
  "",
]
`;

exports[`coordinates > matching > decimal degrees > separator space: 40.689247 -74.044502 1`] = `
[
  "",
  "",
  "40",
  ".689247",
  "",
  "",
  "-",
  "74",
  ".044502",
  "",
]
`;
